### PR TITLE
Create `<CodeBlock />` spec

### PR DESCRIPTION
**Pull request by stephenjwatkins**
## 📝 Changes

- Adds CodeBlock spec

### CODE REVIEW

File: documentation/specs/CodeVisualizer.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# `CodeVisualizer` Component Specification
+
+## Design
```
Comment (stephenjwatkins) [inline documentation/specs/CodeVisualizer.md:3]: the design for this was heavily inspired by Twilio's: https://paste.twilio.design/components/code-block#tabs

File: documentation/specs/CodeVisualizer.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+`CodeVisualizer` wraps one or more `CodeBlock`s with a header to allow the user to switch between blocks.
+
+`CodeVisualizer` will be a controlled component. It will require the consumer to provide a selected language and manage its state.
```
Comment (stephenjwatkins) [inline documentation/specs/CodeVisualizer.md:7]: how do we feel about it being only controlled? only controlled simplifies our implementation but may be a little more friction for the consumer. alternatively we could implement for it allowing both controlled and uncontrolled at the cost of a little more implementation complexity

File: documentation/specs/CodeVisualizer.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+### API
+
+```ts
+export type CodeVisualizerProps = {
+  /**
+   * CodeVisualizer content. This should be a header and collection of snippets.
+   */
+  children: ReactNode;
```
Comment (stephenjwatkins) [inline documentation/specs/CodeVisualizer.md:16]: if there's only one snippet provided, the CodeVisualizer goes into "View Only" mode and doesn't show a language dropdown

File: documentation/specs/CodeVisualizer.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+   * Code snippet content. This should contain a CodeBlock.
+   */
+  children: ReactNode;
+};
+```
+
+### Example Usage
+
+```tsx
+import { CodeVisualizer } from "@easypost/easy-ui/CodeBlock";
```
Comment (OskiTheCoder) [inline documentation/specs/CodeVisualizer.md:54]: **question**: is the language picker/selector intended to function as a trigger for a menu/listbox? if so, do you have a rough idea for how you want to implement the trigger?

File: documentation/specs/CodeVisualizer.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
+# `CodeVisualizer` Component Specification
```
[the remark this answers is not in the document]
Comment (OskiTheCoder) [reply documentation/specs/CodeVisualizer.md:1]: im ultimately ok with `CodeVisualizer`, but just some thoughts: at first glance when i see `CodeBlock` and `CodeVisualizer`, its not immediately obvious to me which is the _molecule_ level component. `CodeBlockWrapper` feels a bit more descriptive and ive seen it out in the wild too, but it also feels kinda verbose to say/write lol.

File: documentation/specs/CodeVisualizer.md (hunk 2 above)
Comment (OskiTheCoder) [reply documentation/specs/CodeVisualizer.md:7]: im personally a fan of it only being controlled based on how im understanding the use case

File: documentation/specs/CodeVisualizer.md (hunk 4 above)
Comment (stephenjwatkins) [reply documentation/specs/CodeVisualizer.md:54]: yeah it's supposed to trigger our `Menu`. i was imagining using an `UnstyledButton` with a `Text` and `Icon` inside as the `Menu.Trigger`

File: documentation/specs/CodeVisualizer.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ldewald) [reply documentation/specs/CodeVisualizer.md:1]: I agree it's not obvious which is the molecule level. What if we renamed the existing `CodeBlock` to be `CodeHighlighter` (or `SyntaxHighlighter`) and used `CodeBlock` for this one? I'm okay with `CodeVisualizer` as well but throwing this idea out there.

File: documentation/specs/CodeVisualizer.md (hunk 2 above)
Comment (ldewald) [reply documentation/specs/CodeVisualizer.md:7]: > im personally a fan of it only being controlled based on how im understanding the use case

+1

File: documentation/specs/CodeVisualizer.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+```tsx
+import { CodeVisualizer } from "@easypost/easy-ui/CodeBlock";
+
+function Component() {
+  const [language, setLanguage] = useState("javascript");
+  return (
+    <CodeVisualizer language={language} onLanguageChange={setLanguage}>
+      <CodeVisualizer.Header>Header</CodeVisualizer.Header>
+      <CodeVisualizer.Snippet>
```
Comment (ldewald) [inline documentation/specs/CodeVisualizer.md:61]: Curious what this wrapper is for, is there a case where we what anything other than (or in addition to) `CodeBlock`?

**Pull request merged**